### ISSUE TITLE
Pylint not allowing whitespace after type hint

### ISSUE THREAD

**Issue by martmists** [PyCQA/pylint#1543]
### Steps to reproduce
![](https://i.alexflipnote.xyz/d76600.png)

### Current behavior
See screenshot above

### Expected behavior
Pylint not erroring

### pylint --version output
1.7.1

---

Travis started erroring on this only a minute ago, despite similar code existing in the same file which it does not error on.
I'm guessing this is because the functiondef is multiline, but I can't say anything for sure

EDIT: It was seen as correct with `user: discord.Member=None`, which is technically against PEP8. I suggest editing pylint to do the reverse.

**Comment by PCManticore** [comment]
Makes sense

**Comment by AWhetter** [comment]
I don't think this is a duplicate because here the error isn't always raised but #1525 is related.

**Comment by mbdevpl** [comment]
I think it's because the annotation is non-trivial i.e. contains a dot. `user: str = None` doesn't generate an error for me. Therefore I think this might be duplicate of #1429 or maybe even #1525.

Do you have any example of non-trivial annotation that is treated by pylint as it should?

**Comment by mbdevpl** [comment]
I've just run into this with the latest pylint 1.8.1:

```
C:257, 0: No space allowed around keyword argument assignment
                t.Tuple[t.Optional[str], t.Optional[str], t.Optional[int]]] = None,
                                                                            ^ (bad-whitespace)
```

**Comment by mbdevpl** [comment]
Please disregard above comment, this is a different issue.

**Comment by mbdevpl** [comment]
@PCManticore @AWhetter Unless I'm very mistaken, issue presented here is already fixed in currently released pylint 1.7 and 1.8?

**Comment by PCManticore** [comment]
@mbdevpl This is indeed fixed in latest release, thanks!
